tools/mo/openvino/tools/mo/front/kaldi/add_reshape_around_convolution.py

```diff
@@ -24,73 +112,102 @@ class ReplaceConvolutionReshape(FrontReplacementPattern):
-        sp_dim_1 = 1
-        if node.has_valid('patch_stride'):
-            channel_dim = 2
-            sp_dim_2 = 3
-            frame_height = node.patch_stride
-        else:
-            channel_dim = 3
-            sp_dim_2 = 2
-            frame_height = node.height_in
-
-        i_shape = Shape(graph, {'name': node_name + '/Shape'}).create_node()
-        i_shape.in_port(0).connect(node.in_port(0).get_source())
-
-        N, H = node_to_get_shape_value_of_indices(i_shape, [0]), node_to_get_shape_value_of_indices(i_shape, [1])
-
-        div = create_op_with_const_inputs(
-            graph, Div, {1: int64_array([frame_height * node.kernel[1]])}, {'name': node_name + '/div_stride_h'})
-        div.in_port(0).connect(H.out_port(0))
-
-        concat = create_op_with_const_inputs(graph, Concat, {sp_dim_2: int64_array([frame_height]),
-                                                             channel_dim: int64_array([node.kernel[1]])},
-                                             {'name': node_name + '/concat_all_dims', 'in_ports_count': 4, 'axis': 0})
-        concat.in_port(0).connect(N.out_port(0))
-        concat.in_port(sp_dim_1).connect(div.out_port(0))
-
-        reshape_in = Reshape(graph, {'name': node_name + '/reshape_in'}).create_node()
-        reshape_in.in_port(1).connect(concat.out_port(0))
-
-        # change layout from NHWC to NCHW
-        # should be replaced by common Permute logic in future
-        transpose = None
-        if channel_dim == 3 and node.channel_dims == 1:
-            transpose = create_op_node_with_second_input(graph, Transpose, int64_array([0, 3, 1, 2]),
-                                                         {'name': node.name + '/Transpose'}, reshape_in)
-
-        # create Reshape after Convolution
-        reshape_out = create_op_node_with_second_input(graph, Reshape, int64_array([0, -1]),
-                                                       {'name': node_name + '/reshape_out'})
-
-        # connect input_reshape_node
-        source = node.in_port(0).get_source()
-        node.in_port(0).get_connection().set_source(transpose.out_port(0) if transpose else reshape_in.out_port(0))
-        reshape_in.in_port(0).connect(source)
-        # connect output_reshape_node
-        node.out_port(0).get_connection().set_source(reshape_out.out_port(0))
-        node.out_port(0).connect(reshape_out.in_port(0))
+    def find_and_replace_pattern(graph: Graph):
+        conv_pool_nodes = graph.get_op_nodes(op='Convolution')
+        conv_pool_nodes.extend(graph.get_op_nodes(op='Pooling'))
+
+        if len(conv_pool_nodes) == 0:
+            return
+
+        set_time_dim(graph)
+
+        update_time_dim_for_start_convolution(graph)
+
+        for node in conv_pool_nodes:
+            node_name = node.soft_get('name', node.id)
+
+            # create Reshape before convolution
+            # shape = [in_shape[0], t, patch_stride, C= in_shape[1]/(patch_stride*t)]
```

Comment:
nit: space is missing `C = in_shape[1]/(patch_stride*t)`

Reply:
done